### PR TITLE
Softfail sshd service failure under FIPS

### PR DESCRIPTION
**Pull request by ge0r**
As mentioned in poo#97016 sshd fails to start in 15 SP2 when FIPS is enabled. A bug has been created (bsc#1189534) and the sshd openqa module will now soft_fail when fialing to start under enabled FIPS.

- Related ticket: https://progress.opensuse.org/issues/97016
- Needles: No needles
- Verification run: [softfailing with FIPS](http://angmar.suse.de/tests/3757#step/sshd/9) | [passing without FIPS](http://angmar.suse.de/tests/3758#)

### CODE REVIEW

File: tests/console/sshd.pm [hunk 1]
```diff
@@ -70,7 +70,15 @@ sub run {
     }
 
     # Restart sshd and check it's status
-    systemctl 'restart sshd';
+    my $ret          = systemctl('restart sshd', ignore_failure => 1);
+    my $fips_enabled = script_output('cat /proc/sys/crypto/fips_enabled');
```
Comment (grisu48) [inline tests/console/sshd.pm:74]: Why not make this a boolean style variable directly here?

```suggestion
    my $fips_enabled = script_output('cat /proc/sys/crypto/fips_enabled') eq '1';
```

And then

```
if (($ret != 0) && ($fips_enabled)) {
```

**Review by grisu48 — APPROVED** [review]
Minor style suggestions, otherwise LGTM :rocket:

File: tests/console/sshd.pm (hunk 1 above)
Comment (ge0r) [reply tests/console/sshd.pm:74]: Much more elegant!

**Review by jouyingbin — COMMENTED** [review]
LGTM, thanks for the change.

**Comment by punkioudi** [comment]
It breaks Leap 15.2 Updates : https://openqa.opensuse.org/tests/overview?distri=opensuse&version=15.2&build=20210819-1&groupid=69

**Comment by punkioudi** [comment]
Also, here : https://openqa.opensuse.org/tests/1879502, https://openqa.opensuse.org/tests/1879508, https://openqa.opensuse.org/tests/1879498, https://openqa.opensuse.org/tests/1879497

**Comment by ge0r** [comment]
Followup PR for systems without the `/proc/sys/crypto/fips_enabled` file: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13095

File: tests/console/sshd.pm [hunk 2]
```diff
@@ -70,7 +70,15 @@ sub run {
     }
 
     # Restart sshd and check it's status
-    systemctl 'restart sshd';
+    my $ret          = systemctl('restart sshd', ignore_failure => 1);
+    my $fips_enabled = script_output('cat /proc/sys/crypto/fips_enabled') eq '1';
+
+    # If restarting sshd service is not successful and fips is enabled, we have encountered bsc#1189534
+    if (($ret != 0) && ($fips_enabled)) {
```
Comment (Vogtinator) [inline tests/console/sshd.pm:77]: Soft failing a critical test like this should IMO be restricted to specific versions. So I suggest

`if (($ret != 0) && $fips_enabled && is_sle("<=15-SP2")) {`

Comment (ge0r) [reply tests/console/sshd.pm:77]: good point

Comment (ge0r) [reply tests/console/sshd.pm:77]: PR that addresses this point: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13096

**Pull request merged**